jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/UpdateCheckerTest.java

```diff
@@ -0,0 +1,198 @@
+
+  private TestWebServer testWebServer;
+  private Path configDir;
+
+  @Before
+  public void setUp()
+      throws InterruptedException, GeneralSecurityException, URISyntaxException, IOException {
+    testWebServer =
+        new TestWebServer(
+            false, Collections.singletonList("HTTP/1.1 200 OK\nContent-Length:5\n\n2.0.0"), 1);
+    configDir = temporaryFolder.getRoot().toPath();
+  }
+
+  @After
+  public void tearDown() throws IOException {
+    testWebServer.close();
+  }
+
+  @Test
+  public void testPerformUpdateCheck_newVersionFound() throws IOException {
+    Instant before = Instant.now();
+    setupConfigAndLastUpdateCheck();
+    Optional<String> message =
+        UpdateChecker.performUpdateCheck(false, "1.0.2", testWebServer.getEndpoint(), configDir);
+    Assert.assertTrue(message.isPresent());
+    Assert.assertEquals(
+        "A new version of Jib (2.0.0) is available (currently using 1.0.2). Update your build "
+            + "configuration to use the latest features and fixes!",
+        message.get());
+    String modifiedTime =
+        new String(
+            Files.readAllBytes(configDir.resolve("lastUpdateCheck")), StandardCharsets.UTF_8);
+    Assert.assertTrue(Instant.parse(modifiedTime).isAfter(before));
+  }
+
+  @Test
+  public void testPerformUpdateCheck_onLatest() throws IOException {
+    Instant before = Instant.now();
+    setupConfigAndLastUpdateCheck();
+    Optional<String> message =
+        UpdateChecker.performUpdateCheck(false, "2.0.0", testWebServer.getEndpoint(), configDir);
+    Assert.assertFalse(message.isPresent());
+    String modifiedTime =
+        new String(
+            Files.readAllBytes(configDir.resolve("lastUpdateCheck")), StandardCharsets.UTF_8);
+    Assert.assertTrue(Instant.parse(modifiedTime).isAfter(before));
+  }
+
+  @Test
+  public void testPerformUpdateCheck_noConfigOrLastUpdateCheck() throws IOException {
+    Instant before = Instant.now();
+    Optional<String> message =
+        UpdateChecker.performUpdateCheck(false, "1.0.2", testWebServer.getEndpoint(), configDir);
+    Assert.assertTrue(message.isPresent());
+    Assert.assertEquals(
+        "A new version of Jib (2.0.0) is available (currently using 1.0.2). Update your build "
+            + "configuration to use the latest features and fixes!",
+        message.get());
+    String modifiedTime =
+        new String(
+            Files.readAllBytes(configDir.resolve("lastUpdateCheck")), StandardCharsets.UTF_8);
+    Assert.assertTrue(Instant.parse(modifiedTime).isAfter(before));
+  }
+
+  @Test
+  public void testPerformUpdateCheck_lastUpdateCheckTooSoon() throws IOException {
+    FileTime modifiedTime = FileTime.from(Instant.now().minusSeconds(12));
+    setupConfigAndLastUpdateCheck();
+    Files.write(
+        configDir.resolve("lastUpdateCheck"),
+        modifiedTime.toString().getBytes(StandardCharsets.UTF_8));
+    Files.setLastModifiedTime(configDir.resolve("lastUpdateCheck"), modifiedTime);
```

Comment:
Is this necessary? If not, remove it, because it's confusing?

Reply:
Oops, old stuff I didn't clean up.